lv_draw/lv_draw_rbasic.c

```diff
@@ -302,21 +303,19 @@ void lv_rmap(const lv_area_t * cords_p, const lv_area_t * mask_p,
     } else {
         lv_color_t chroma_key_color = LV_COLOR_TRANSP;
         lv_coord_t col;
+        bzero(pixel_buffer, sizeof(pixel_buffer));
         for(row = masked_a.y1; row <= masked_a.y2; row++) {
             for(col = masked_a.x1; col <= masked_a.x2; col++) {
                 lv_color_t * px_color = (lv_color_t *) &map_p[(uint32_t)(col - masked_a.x1) * sizeof(lv_color_t)];
-
                 if(chroma_key && chroma_key_color.full == px_color->full) continue;
-
                 if(recolor_opa != LV_OPA_TRANSP) {
                     lv_color_t recolored_px = lv_color_mix(recolor, *px_color, recolor_opa);
-
-                    lv_rpx(col, row, mask_p, recolored_px, LV_OPA_COVER);
+                    pixel_buffer[col - masked_a.x1] = recolored_px;
                 } else {
-                    lv_rpx(col, row, mask_p, *px_color, LV_OPA_COVER);
+                    pixel_buffer[col - masked_a.x1] = *px_color;
```

Comment:
There are no bound checks here. `col - masked_a.x1` will overflow the pixel buffer. You need to copy the image into the buffer in "chunks" and draw the buffer once each chunk is filled.